File: packages/health_connector_hc_android/android/src/main/kotlin/com/phamtunglam/health_connector_hc_android/mappers/health_measurement_unit_mappers/NumberMapper.kt
package com.phamtunglam.health_connector_hc_android.mappers.health_measurement_unit_mappers

import com.phamtunglam.health_connector_hc_android.pigeon.NumberDto

/**
 * Converts a numeric value to a [NumberDto].
 */
internal fun Number.toNumberDto(): NumberDto = NumberDto(value = this.toDouble())

File: packages/health_connector_hc_android/lib/src/mappers/measurement_unit_mappers/number_mapper.dart
import 'package:health_connector_core/health_connector_core_internal.dart'
    show Number;
import 'package:health_connector_hc_android/src/pigeon/health_connector_hc_android_api.g.dart'
    show NumberDto;
import 'package:meta/meta.dart' show internal;

/// Converts [Number] to [NumberDto].
@internal
extension NumberToDto on Number {
  NumberDto toDto() {
    return NumberDto(
      value: value.toDouble(),
    );
  }
}

/// Converts [NumberDto] to [Number].
@internal
extension NumberDtoToDomain on NumberDto {
  Number toDomain() {
    return Number(value);
  }
}
